### PR TITLE
Added optional parameter 'freeform_tags'

### PR DESCRIPTION
**Pull request by slippman**

### CODE REVIEW

**Comment by stephenpearson** [comment]
Thanks for this @slippman.  For completeness, could you also add the following mandatory freeform tags?
- kitchen = true
- cookbook_name = <cookbook_name>
- cookbook_version = <cookbook_version>

I think you may be able to get the cookbook name and version from either the state or the instance objects.  You might have to dump those out to see.

**Comment by slippman** [comment]
I noticed that if a hostname_prefix was not supplied, it created a problem due to the insertion of the '-' delimiter. I have added code to address that. I am also hard coding the creation of a `kitchen` freeform tag, and a run_list or policyfile tag if those are available on the instance object. Other user supplied freeform tags will be set as well. For example: 

```
platforms:
  - name: oracle-7
    driver:
      hostname_prefix: gta-slippman
      freeform_tags:
        cookbook: generator_test
```

As I was not able to find a way to get the cookbook name from the instance or state object. 

Screen Shot 2019-07-09 at 11 47 45 PM

**Comment by stephenpearson** [comment]
That looks excellent @slippman.  I hope you don't mind, but I pushed up a few changes to your branch to help the code pass all Rubocop linting checks and to improve readability slightly.

I've run a few tests and the tags appear to work along with the fixed hostname prefix.  I'll give you a chance to review the changes I made to your branch, and if you're happy then I'll merge it and publish the gem into rubygems.org.

File: lib/kitchen/driver/oci.rb [hunk 1]
```diff
@@ -55,9 +57,9 @@ class Oci < Kitchen::Driver::Base # rubocop:disable Metrics/ClassLength
 
       def process_freeform_tags(freeform_tags)
         prov = instance.provisioner.instance_variable_get(:@config)
-        tags = ['run_list' , 'policyfile']
+        tags = %w[run_list policyfile]
         tags.each do |tag|
-          if prov[tag.to_sym] != nil && prov[tag.to_sym].empty? == false
+          unless prov[tag.to_sym].nil? || prov[tag.to_sym].empty?
```
Comment (slippman) [inline lib/kitchen/driver/oci.rb]: Ah, this is much cleaner, thanks!

File: lib/kitchen/driver/oci.rb [hunk 2]
```diff
@@ -269,29 +271,27 @@ def inject_powershell(state)
...
-        request = OCI::Core::Models::LaunchInstanceDetails.new
+      def generate_hostname
         prefix = config[:hostname_prefix]
-        if prefix.empty?
-          hostname = random_hostname(instance.name)
-        else
-          hostname = random_hostname(prefix + '-' + instance.name)
-        end
+        [prefix, random_hostname(instance.name)].compact.join('-')
+      end
```
Comment (slippman) [inline lib/kitchen/driver/oci.rb]: ` useful method if you need to remove nil values from an array is compact` - hey, thats neat!

**Comment by slippman** [comment]
@stephenpearson , changes look good. Im gonna compile it locally and use it today, and if everything still seems to be working, i'll let ya know here... Thanks! Code looks cleaner now!

**Comment by stephenpearson** [comment]
@slippman Great, thank you.

File: lib/kitchen/driver/oci.rb (hunk 2 above)
Comment (stephenpearson) [reply lib/kitchen/driver/oci.rb]: We'll make a Rubyist out of you yet :)

**Comment by slippman** [comment]
@stephenpearson Testing completed.. looks good! I'm good with a merge..

**Comment by stephenpearson** [comment]
Sounds good @slippman, I'll merge it.

**Pull request merged**